packages/documentation/src/demo-components/demo-button.ts

```diff
@@ -0,0 +1,83 @@
+export class DemoButton extends HTMLElement {
+  static get observedAttributes() {
+    return ['button-version', 'workaround', 'arialabelledby-id', 'ariadescribedby-id'];
+  }
+  private buttonVersion?: '1' | '2' | '3' | '4';
+  private workaround?: string;
+  private internalButton?: HTMLElement;
+  private arialabelledbyId?: string;
+  private ariadescribedbyId?: string;
+  private slotEl?: HTMLSlotElement;
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+  }
+
+  connectedCallback() {
+    this.render();
+  }
+
+  attributeChangedCallback(name: string, _oldValue: string, newValue: string) {
+    if (name === 'workaround') this.workaround = newValue;
+    if (name === 'arialabelledby-id') this.arialabelledbyId = newValue;
+    if (name === 'ariadescribedby-id') this.ariadescribedbyId = newValue;
+    if (name === 'button-version') this.buttonVersion = newValue as '1' | '2' | '3';
+    this.render();
+  }
```

Comment:
Since only the observed attributes will execute this callback, you could easily make it generic.

```ts
attributeChangedCallback(name: string, _oldValue: string, newValue: string) {
  const member = name.replace(/-./g, m => m[1].toUpperCase());
  
  this[member] = newValue;
  this.render();
}
```